lib/handlers/PostHandler.ts

```diff
@@ -68,10 +68,27 @@ export default class PostHandler extends BaseHandler {
     const url = this.generateUrl(req, file.id)
     this.emit(EVENTS.EVENT_ENDPOINT_CREATED, {url})
 
-    const optional_headers: {'Upload-Offset'?: string} = {}
+    const optional_headers: {
+      'Upload-Offset'?: string
+      'Upload-Expires'?: string
+    } = {}
+
+    const _addUploadExpiresHeader = (store: DataStore) => {
+      // If expiration is known at creation time, Upload-Expires header MUST be included in the response
+      if (store.hasExtension('expiration') && this.store.getExpiration() > 0) {
+        const creation = new Date(file.creation_date as Date)
+        // Value MUST be in RFC 7231 datetime format
+        optional_headers['Upload-Expires'] = new Date(
+          creation.getTime() + this.store.getExpiration() * 60_000
+        ).toUTCString()
+      }
+    }
 
-    // The request MIGHT include a Content-Type header when using creation-with-upload extension
-    if (!RequestValidator.isInvalidHeader('content-type', req.headers['content-type'])) {
+    if (RequestValidator.isInvalidHeader('content-type', req.headers['content-type'])) {
```

Comment:
I would leave this as it was, also with the deleted comment, and start a new block specifically for expiry. I find this flow control hard to follow.